src/metabase/embed/settings.clj

```diff
@@ -25,6 +27,8 @@
   :setter     (fn [new-value]
                 (when (not= new-value (setting/get-value-of-type :boolean :enable-embedding))
                   (setting/set-value-of-type! :boolean :enable-embedding new-value)
+                  (when (and new-value (str/blank? (setting/get :embedding-secret-key)))
+                    (setting/set! :embedding-secret-key (crypto-random/hex 32)))
```

Comment:
In general we prefer using the automatically-generated getter/setter functions for settings which are the same as the setting name, with a `!` suffix for the setter.
```suggestion
                  (when (and new-value (str/blank? (embedding-secret-key)))
                    (embedding-secret-key! (crypto-random/hex 32)))
```

(This assumes that `embedding-secret-key` is defined in the same namespace... if not you'll have to prefix it with the namespace as well.)

Reply:
Thank you @noahmoss, the IDE suggested to use :refer when importing, but I saw in other places we usually just import the namespace and do namespace/function so that' what I did in dca0965d8b87ff89d25ae7de407087efe8b4212a

Reply:
Yeah, looks good